src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Binders/PlaceholderLocalBinder.cs

```diff
@@ -123,6 +125,22 @@ private PlaceholderLocalSymbol LookupPlaceholder(string name)
             var type = typeNameDecoder.GetTypeSymbolForSerializedType(typeName);
             Debug.Assert((object)type != null);
 
+            var dynamicFlagsInfo = alias.CustomTypeInfo.ToDynamicFlagsCustomTypeInfo();
+            if (dynamicFlagsInfo.Any())
+            {
+                var flagsBuilder = ArrayBuilder<bool>.GetInstance();
+                dynamicFlagsInfo.CopyTo(flagsBuilder);
+                var dynamicType = DynamicTypeDecoder.TransformTypeWithoutCustomModifierFlags(
```

Comment:
My suggestion to use BitArray originally was probably overkill. Would this be simpler if there were just two representations of the bits, as byte[] and bool[], with (static) methods to convert between them, much as you had previously?